webxrlayers-1.bs

```diff
@@ -1386,6 +1386,53 @@ when the [=XR Compositor=] can no longer reproject the layer.
 
 The author SHOULD redraw the content of the layer at the next [=XR animation frame=]. The event must be of type {{XRLayerEvent}}.
 
+Hit testing {#hittesting}
+===========
+
+XRLayerHitResult {#xrlayerhitresulttype}
+----------------
+
+<pre class="idl">
+dictionary XRLayerHitResult {
+  XRCompositionLayer layer;
+  unsigned long x;
+  unsigned long y;
+  DOMPointInit position;
+};
+</pre>
+
+The <dfn dict-member for="XRLayerHitResult">layer</dfn> member defines what layer was hit.
+
+The <dfn dict-member for="XRLayerHitResult">x</dfn> and <dfn dict-member for="XRLayerHitResult">y</dfn> member define the x and y position
+where the layer was hit.
+
+The <dfn dict-member for="XRLayerHitResult">position</dfn> member defines the position in 3D space where the hit occured.
+
+Finding hit tests {#finding-hit-tests}
+-----------------
+
+<pre class="idl">
+[SecureContext, Exposed=Window] partial interface XRSession {
+  XRLayerHitResult? findHitTest(XRRigidTransform ray);
```

Comment:
This should be on the frame, and should probably return a sequence or something.

Also, returning dictionaries isn't good, we should return an interface.

Also, we should probably use XRRay here, and pass in an XRSpace as a base space.

Reply:
> This should be on the frame, and should probably return a sequence or something.

Why should it be on the frame? The state of the layers is managed by the session and determined by the layers array.

I also don't see a need to return a sequence. Only 1 layer can be hit at a time.

> Also, returning dictionaries isn't good, we should return an interface.

OK. I will fix that.

> Also, we should probably use XRRay here, and pass in an XRSpace as a base space.

I didn't use XRRay because I'm unsure when that spec will transition to the WG and I don't want to reference something that might not become official.

Reply:
> Why should it be on the frame? The state of the layers is managed by the session and determined by the layers array.

XRRigidTransforms are meaningless without a space to relate them to (which you've added, thanks).

The position of a space can vary over time. This needs to be per-frame to account for that.

> I also don't see a need to return a sequence. Only 1 layer can be hit at a time.

I'm not so sure of that, what if I have a transparentish HUD layer that I want selection to pass through, so that it hits the actual layer I care about?

> I didn't use XRRay because I'm unsure when that spec will transition to the WG and I don't want to reference something that might not become official.

I believe hit-test was already proposed for movement into the WG before layers was, so either way it won't move to WG _after_ layers does.

Reply:
> > Why should it be on the frame? The state of the layers is managed by the session and determined by the layers array.
> 
> XRRigidTransforms are meaningless without a space to relate them to (which you've added, thanks).
> 
> The position of a space can vary over time. This needs to be per-frame to account for that.

The hit test API is on `Session`. Are you saying that is implemented in the wrong location as well?

> > I also don't see a need to return a sequence. Only 1 layer can be hit at a time.
> 
> I'm not so sure of that, what if I have a transparentish HUD layer that I want selection to pass through, so that it hits the actual layer I care about?

hmm, I'm unsure. Maybe if we define that the first layer that is hit, is also the first one in the array.

> > I didn't use XRRay because I'm unsure when that spec will transition to the WG and I don't want to reference something that might not become official.
> 
> I believe hit-test was already proposed for movement into the WG before layers was, so either way it won't move to WG _after_ layers does.

We had a small discussion on this on last week's call. The Google team didn't ask for it to move to the WG yet.
Also, it's a bit weird that we define a whole new interface for something that has an almost identical interface to a XRRigidTransform...

Reply:
> The hit test API is on Session. Are you saying that is implemented in the wrong location as well?

The hit test API has a _subscriber_ API on Session, actual poses are queried off of the Frame.

There is no subscriber API here, coordinates are being directly queried.

> We had a small discussion on this on last week's call. The Google team didn't ask for it to move to the WG yet.

Oh, my bad, I thought they had.

>  Also, it's a bit weird that we define a whole new interface for something that has an almost identical interface to a XRRigidTransform...

XRRigidTransform overspecifies a ray and kinda represents a different concept. But yes, they are quite similar.

Reply:
> > The hit test API is on Session. Are you saying that is implemented in the wrong location as well?
> 
> The hit test API has a _subscriber_ API on Session, actual poses are queried off of the Frame.

OK. I will update the PR.